### PR TITLE
Additional `BlackboxBoundedContext` tests

### PR DESCRIPTION
**Pull request by serhii-lekariev**
[This PR](https://github.com/SpineEventEngine/core-java/pull/976) introduced method to register event dispatchers with `BlackBoxBoundedContext`.

This PR follows up on those changes by adding a method to register command dispatchers, as well as testing both registering event and command dispatchers.

### CODE REVIEW

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

**Comment by serhii-lekariev** [comment]
@armiol, I think the build should pass once https://github.com/SpineEventEngine/core-java/pull/981 is merged.

File: testutil-server/src/test/java/io/spine/testing/server/blackbox/BlackBoxBoundedContextTest.java [hunk 1]
```diff
@@ -106,7 +112,66 @@ void tearDown() {
     T boundedContext() {
         return context;
     }
-    
+
+    @Test
+    @DisplayName("register command dispatchers")
+    void registerCommandDispatchers() {
+        CommandClass commandTypeToIntercept = CommandClass.from(BbRegisterCommandDispatcher.class);
```
Comment (armiol) [inline testutil-server/src/test/java/io/spine/testing/server/blackbox/BlackBoxBoundedContextTest.java]: `commandTypeToDispatch`, as we do not "intercept" commands.

Same below.

File: testutil-server/src/test/java/io/spine/testing/server/blackbox/BlackBoxBoundedContextTest.java [hunk 2]
```diff
@@ -106,7 +112,66 @@ void tearDown() {
...
+    }
+
+    @Test
+    @DisplayName("throw on an attempt to register duplicate command dispatchers")
+    void throwOnDuplicateCommandDispatchers() {
+        CommandClass commandTypeToIntercept = CommandClass.from(BbRegisterCommandDispatcher.class);
+        BbCommandDispatcher dispatcher = new BbCommandDispatcher(context.eventBus(),
+                                                                 commandTypeToIntercept);
+        context.withHandlers(dispatcher);
+        assertThrows(IllegalArgumentException.class, () -> context.withHandlers(dispatcher));
```
Comment (armiol) [inline testutil-server/src/test/java/io/spine/testing/server/blackbox/BlackBoxBoundedContextTest.java]: Instead, let's try to register a dispatcher of a different type which handles the same command.

File: testutil-server/src/test/proto/spine/testing/server/blackbox/events.proto [hunk 3]
```diff
@@ -66,6 +66,11 @@ message BbAssigneeRemoved {
     spine.core.UserId user_id = 2 [(required) = true];
 }
 
+message EventDispatcherRegistered {
```
Comment (armiol) [inline testutil-server/src/test/proto/spine/testing/server/blackbox/events.proto]: `BbEventDispatcherRegistered`

**Comment by serhii-lekariev** [comment]
@armiol, PTAL again.

File: testutil-server/src/test/java/io/spine/testing/server/blackbox/given/BbDuplicateCommandDispatcher.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+import java.util.Set;
+
+/**
+ * A command dispatcher that dispatches the same type of commands as
+ * {@link io.spine.testing.server.blackbox.given.BbCommandDispatcher}.
+ *
+ * <p>Attempting to register them with the same bounded context should result in an exception.
+ */
+
```
Comment (armiol) [inline testutil-server/src/test/java/io/spine/testing/server/blackbox/given/BbDuplicateCommandDispatcher.java]: Why an empty line?

**Pull request merged**